package.json

```diff
@@ -63,8 +63,8 @@
     "@types/fs-extra": "^11.0.1",
     "@types/node": "^18.15.11",
     "@types/prettier": "^2.7.3",
-    "@typescript-eslint/eslint-plugin": "^5.58.0",
-    "@typescript-eslint/parser": "^5.58.0",
+    "@typescript-eslint/eslint-plugin": "^6.18.1",
+    "@typescript-eslint/parser": "^6.18.1",
```

Comment:
This is to match the versions here https://github.com/aws-amplify/amplify-backend/blob/f2c5c47812c404bc225423fd2308fcc94f9f4a4b/packages/eslint-rules/package.json#L7-L12
Npm was hoisting `6.18.1` node modules in the scope of our `eslint-rules` package causing the `Rule` to come from locally hoisted module and `RuleTester` from global node_modules creating conflict.